code/datums/helper_datums/teleport.dm

```diff
@@ -238,6 +285,26 @@
 	if(do_teleport(affected_mob, pick(L), channel = TELEPORT_CHANNEL_MAGIC, no_effects = TRUE))
 		affected_mob.say("SCYAR NILA [uppertext(thearea.name)]!", forced = "wizarditis teleport")
 
+// This is here because I believe it makes the most sense here.
+/mob/living/proc/handle_teleport_interception(datum/source, channel, turf/origin, turf/destination)
+	SIGNAL_HANDLER
+
+	if(channel != TELEPORT_CHANNEL_GATEWAY)
+		return COMPONENT_ALLOW_TELEPORT
```

Comment:
COMPONENT_ALLOW_TELEPORT implies that we want to ignore other teleport restrictions entirely. I think this should just return no value in order to indicate that we want the signal caller to decide if teleportation should be allowed and that we want to have no effect.

Reply:
Removed COMPONENT_ALLOW_TELEPORT